packages/contracts/src/PlayerHandle.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Ownable} from "openzeppelin/access/Ownable.sol";
+import {Strings} from "openzeppelin/utils/Strings.sol";
+
+interface IENSResolver {
+    function getEnsName(address addr) external view returns (string memory);
```

Comment:
That doesn't seem to be the actual ENS resolver interface, which uses `name()` for lookups, see here: https://docs.ens.domains/resolvers/public

We also probably won't deploy on Ethereum, which is the only chain that has this resolver at the moment. ENS has announced that L2 support is coming but it's not there yet. I think it's fine to keep the resolver interface for now, at worst we can stub it with a permissioned registry that we sign on — let's just make the interface the same as the real thing.